docs/architecture/adr-043-nft-module.md

```diff
@@ -0,0 +1,413 @@
+  rpc Burn(MsgBurn)   returns (MsgBurnResponse);
+}
+
+message MsgIssue {
+  cosmos.nft.v1beta1.Metadata metadata = 1;
+  string issuer                        = 2;
+}
+message MsgIssueResponse {}
+
+message MsgMint {
+  cosmos.nft.v1beta1.NFT nft = 1;
+  string minter              = 2;
+}
+message MsgMintResponse {}
+
+message MsgEdit {
+  cosmos.nft.v1beta1.NFT nft = 1;
+  string editor              = 2;
+}
+message MsgEditResponse {}
+
+message MsgSend {
+  string type     = 1;
+  string id       = 2;
+  string sender   = 3;
+  string reveiver = 4;
+}
+message MsgSendResponse {}
+
+message MsgBurn {
+  string type      = 1;
+  string id        = 2;
+  string destroyer = 3;
+}
+message MsgBurnResponse {}
+```
+
+`MsgIssue` can be used to issue an NFT type/class, just like deploying an ERC721 contract on Ethereum.
+
+`MsgMint` allows users to create new NFTs for a given type.
+
+`MsgEdit` allows users to edit/update their NFTs.
+
+`MsgSend` can be used to transfer the ownership of an NFT to another address.
+**Note**: we could use `x/bank` to handle NFT transfer directly and do without this service. It's only for the sake of completeness of an ERC721 compatible API that we may choose to keep this service.
+
+`MsgBurn` allows users to destroy their NFTs.
+
+Other business logic implementations should be defined in other upper-level modules that import this NFT module. The implementation example of the server is as follows:
+
+```go
+type msgServer struct{
+  k Keeper
+}
+
+func (m msgServer) Issue(ctx context.Context, msg *types.MsgIssue) (*types.MsgIssueResponse, error) {
+  m.keeper.AssertTypeNotExist(msg.Metadata.Type)
+
+  bz := m.keeper.cdc.MustMarshalBinaryBare(msg.Metadata)
+  typeStore := m.keeper.getTypeStore(ctx)
+  typeStore.Set(msg.Type, bz)
+  
+  bz := m.keeper.cdc.MustMarshalBinaryBare(msg.Issuer)
+  typeOwnerStore := m.keeper.getTypeOwnerStore(ctx)
+  typeOwnerStore.Set(msg.Type, bz)
+  
+  return &types.MsgIssueResponse{}, nil
+}
+
+func (m msgServer) Mint(ctx context.Context, msg *types.MsgMint) (*types.MsgMintResponse, error) {
+  m.keeper.AssertTypeExist(msg.NFT.Type)
+  m.keeper.AssertCanMint(msg.NFT.Type, msg.Minter)
+  
+  baseDenom := fmt.Sprintf("%s-%s", msg.NFT.Type, msg.NFT.Id)
+  bkMetadata := bankTypes.Metadata{
+    Symbol:      metadata.Symbol,
+    Base:        baseDenom,
+    Name:        metadata.Name,
+    URI:         msg.NFT.URI,
+    Description: metadata.Description,
+  }
+  
+  m.keeper.bank.SetDenomMetaData(ctx, bkMetadata)
+  mintedCoins := sdk.NewCoins(sdk.NewCoin(baseDenom, 1))
+  m.keeper.bank.MintCoins(types.ModuleName, mintedCoins)
+  m.keeper.bank.SendCoinsFromModuleToAccount(types.ModuleName, msg.Owner, mintedCoins)
+  
+  bz := m.keeper.cdc.MustMarshalBinaryBare(&msg.NFT)
+  
+  nftStoreByType := m.keeper.getNFTStoreByType(ctx, msg.NFT.Type)
+  nftStoreByType.Set(msg.NFT.Id, bz)
+  
+  nftStore := m.keeper.getNFTStore(ctx)
+  nftStore.Set(baseDenom, bz)
+  
+  return &types.MsgMintResponse{}, nil
```

Comment:
Do we need to provide the implementation details on the ADR level? How about using some pseudo-code? cc: @aaronc 

Eg:
```
Create denom = `type-id`
Create and bank.Metatadata using bank keeper
create a coin with total supply = 1 using bank keeper
create NFT' from msg.NFT
store NFT': nftStore.Set(denom, nft')
```